### PR TITLE
Migration: Make sure to delete WorkloadEndpoint on old host

### PR DESCRIPTION
**Pull request by nelljerram**
update_port_postcommit processing could skip calling _migration_step,
even when an actual migration was occurring, because of preceding
checks for whether the port was bound before and after the update.
Skipping _migration_step means that the WorkloadEndpoint on the old
host doesn't get deleted - at least until the next resync - and that
leaves a problem if there is a second migration of the same port back
to the old host...

In the event of a second migration of the same port back to the old
host, the problem is as follows.

- The WorkloadEndpoint still exists on the old host.  (As well as on
  the new host; these are distinct WorkloadEndpoints, even though they
  correspond to the same Neutron port at different times of its life.)

- Nova recreates the tap interface, on the old host, very early on
  during the second migration process.  (So that it is possible to
  create a new VM there and start migrating to it.)

- As soon as the interface exists, as well as the WorkloadEndpoint,
  Felix (+ BIRD) starts advertising out that the WorkloadEndpoint's IP
  exists on that host.

- But in fact this is way too early, because the new VM isn't yet
  ready, and it causes problems that the same IP is being advertised out
  from both the old and the new host.

To eliminate that problem, it is important to delete the
WorkloadEndpoint on the old host when the process of migration to a
new host completes.  `_migration_step` was designed to do this, but it
wasn't being hit (at least in Rocky and Ussuri) because we were
instead hitting this prior if condition:

     elif port_bound(port) and not port_bound(original):
         LOG.info("Port becoming bound: create.")
         self.endpoint_syncer.write_endpoint(port,
                                             context._plugin_context)

In other words, in recent OpenStack releases, it appears that the
`original` port is marked as being unbound, in the
update_port_postcommit callback that also updates `binding:host_id`.

To fix that, and generally try to make the update logic more robust:

- Check upfront for a `binding:host_id` change, independent of port
  bound state, and delete the old WorkloadEndpoint in that case.

- Eliminate the `_migration_step` code and case, as it now reduces to
  just creating or updating the new WorkloadEndpoint, and that is
  already handled by other cases.

- Comment in detail about why we re-read the port data.

- Use `port_unbound` utility function in `_update_port`, instead of
  its own equivalent.

- Remove unnecessary commenting in `_update_port`.

### CODE REVIEW

File: networking_calico/plugins/ml2/drivers/calico/mech_calico.py [hunk 1]
```diff
@@ -779,25 +811,24 @@ def update_port_postcommit(self, context):
...
+            elif port_bound(port):
+                if endpoint_should_already_exist:
+                    LOG.info("Port update")
+                    self.endpoint_syncer.write_endpoint(port,
+                                                        plugin_context,
+                                                        must_update=True)
+                else:
+                    LOG.info("Port becoming bound: create.")
+                    self.endpoint_syncer.write_endpoint(port,
+                                                        context._plugin_context)
```
Comment (fasaxc) [inline networking_calico/plugins/ml2/drivers/calico/mech_calico.py]: Use local `plugin_context` variable for symmetry with above case?

**Review by fasaxc — APPROVED** [review]
LGTM with one nit
